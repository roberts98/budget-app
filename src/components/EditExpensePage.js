import React from 'react';
import { connect } from 'react-redux';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

const EditExpensePage = (props) => {
  return (
    <div>
      <div className="page-header">
        <div className="container">
          <h1 className="page-header__title">Edit Expense</h1>
        </div>
      </div>
      <div className="container">
        <div className="button-flex">
          <ExpenseForm
            expense={props.expense}
            onSubmit={(expense) => {
              props.dispatch(startEditExpense(props.expense.id, expense));
              props.history.push('/');
            }}
          />
          <button
            className="button button--secondary"
            onClick={() => {
              props.dispatch(startRemoveExpense({ id: props.expense.id }));
              props.history.push('/');
            }}
          >
            Remove Expense
      </button>
        </div>
      </div>
    </div>
  );
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
}

export default connect(mapStateToProps)(EditExpensePage);