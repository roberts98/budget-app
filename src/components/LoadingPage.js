import React from 'react';

const LoadingPage = () => (
  <div className="loader">
    <img src="/images/giphy.gif" />
  </div>
);

export default LoadingPage;