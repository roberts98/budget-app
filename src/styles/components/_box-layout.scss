.box-layout {
  align-items: center;
  background: url('/images/bg.jpeg');
  background-size: cover;
  display: flex;
  justify-content: center;
  height: 100vh;
  width: 100vw;
}

.box-layout__box {
  background: fade-out(#e8e8e8, .15);
  border-radius: 5px;
  padding: $l-size;
  text-align: center;
  width: 25rem;
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
}