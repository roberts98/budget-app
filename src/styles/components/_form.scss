.form {
  display: flex;
  flex-direction: column;
  > * {
    margin-bottom: $m-size;
  }
}

.form__error {
  background: #c00;
  border-radius: 10px;
  color: white;
  margin: 0 0 $l-size 0;
  padding: $s-size 0;
  font-style: italic;
  text-align: center;
}