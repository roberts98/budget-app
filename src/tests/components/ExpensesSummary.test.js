import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render expenses sumamary with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={195.20} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render expenses sumamary with multiple expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={395.20} />);
  expect(wrapper).toMatchSnapshot();
});