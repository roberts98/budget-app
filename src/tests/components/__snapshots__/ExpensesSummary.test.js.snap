// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expenses sumamary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $3.95
  </h1>
</div>
`;

exports[`should correctly render expenses sumamary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.95
  </h1>
</div>
`;
